src/components/test-element-settings/TestElementSettings.test.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
+import TestElementSettings from "./TestElementSettings";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { ITestElementSettings } from "@/AppSettings";
+import ElementManager from "@/ElementManager";
+
+const TEST_SETTING: ITestElementSettings = {
+  atomic: 26,
+  enabled: true,
+  stats: {
+    right: 0,
+    times: 0,
+    wrong: 0,
+  },
+};
+
+describe("TestElementSettings", () => {
+  beforeAll(() => {
+    ElementManager.loadElements();
+  });
+
+  it("shows the elements symbol", () => {
+    render(<TestElementSettings setting={TEST_SETTING} />);
+
+    expect(screen.getByText("Fe")).toBeInTheDocument();
+  });
+
+  it("shows the elements atomic number and name", () => {
+    render(<TestElementSettings setting={TEST_SETTING} />);
+
+    expect(screen.getByText("26. Iron")).toBeInTheDocument();
+  });
+
+  it("shows the elements group", () => {
+    render(<TestElementSettings setting={TEST_SETTING} />);
+
+    expect(screen.getByText("Transition metals")).toBeInTheDocument();
+  });
+
+  it("calls onClick when the element is clicked", () => {
+    const onClick = jest.fn();
+    render(<TestElementSettings setting={TEST_SETTING} onClick={onClick} />);
+
+    userEvent.click(screen.getByText("26. Iron"));
+
+    expect(onClick).toHaveBeenCalledWith(26);
```

Comment:
If the button is clicked there is likely UI behavior that can be asserted
1. Assert UI is a way before the click
2. Click on button
3. Assert UI has behaved in an expected way after the click

Reply:
We can't do that (afaik) since this is a "dumb" component, the logic for turning it on/off it's handled in the parent, this component only knows that should call `onClick` when it's pressed and the checkbox state it's based on props too. It doesn't handle the logic of changing the checkbox state.

Reply:
Then it should be an integration test (utilizing the parent) or just tested in the parent. Testing code implementation details, like "did this onClick function get called" is meaningless to the UI/UX and the antithesis of RTL

Reply:
Done!